test/extended/builds/digest.go

```diff
@@ -0,0 +1,106 @@
+package builds
+
+import (
+	"fmt"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[builds][Slow] completed builds should have digest of the image in their status", func() {
+	defer g.GinkgoRecover()
+	var (
+		imageStreamFixture = exutil.FixturePath("..", "integration", "testdata", "test-image-stream.json")
+		stiBuildFixture    = exutil.FixturePath("testdata", "test-s2i-build.json")
+		dockerBuildFixture = exutil.FixturePath("testdata", "test-docker-build.json")
+		oc                 = exutil.NewCLI("build-sti-labels", exutil.KubeConfigPath())
+	)
+
+	g.JustBeforeEach(func() {
+		g.By("waiting for builder service account")
+		err := exutil.WaitForBuilderAccount(oc.AdminKubeClient().Core().ServiceAccounts(oc.Namespace()))
+		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+
+	g.Describe("S2I build", func() {
+		g.Describe("started with normal log level", func() {
+			g.It(fmt.Sprintf("should save the image digest when finished"), func() {
+
+				g.By("creating test imagestream")
+				err := oc.Run("create").Args("-f", imageStreamFixture).Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				g.By("creating test build")
+				err = oc.Run("create").Args("-f", stiBuildFixture).Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				g.By("starting a test build")
+				br, err := exutil.StartBuildAndWait(oc, "test")
+
+				g.By("checking that the resulting image has a digest")
+				o.Expect(br.Build.Status.Output.To.ImageDigest).To(o.HavePrefix("sha256:"))
+			})
+		})
+
+		g.Describe("started with log level >5", func() {
+			g.It(fmt.Sprintf("should save the image digest when finished"), func() {
```

Comment:
All the `It` blocks differ in just few values. Could you make it a top-level parametrised function and call it from each sub-Describe?